container/dangerzone.py

```diff
@@ -181,64 +175,81 @@ def document_to_pixels(self) -> None:
-        run_command(
-            args,
-            error_message="Separating document into pages failed",
-            timeout_message=f"Error separating document into pages, pdftk timed out after {DEFAULT_TIMEOUT} seconds",
-        )
+        self.update_progress("Obtaining PDF metadata")
 
-        page_filenames = glob.glob("/tmp/page-*.pdf")
+        # Obtain number of pages
+        result = run_command(
+            ["pdfinfo", pdf_filename],
+            error_message="PDF file is corrupted",
+            timeout_message=f"Extracting metadata from PDF timed out after {DEFAULT_TIMEOUT} seconds",
+        )
+        output_lines = result.stdout.decode().split("\n")
+        num_pages = None
+        for line in output_lines:
+            if "Pages:" in line:
+                search = re.search(r"\d+", line)
+                if search is not None:
+                    num_pages = int(search.group(0))
+        if num_pages is None:
+            raise ValueError("Number of pages could not be extraced from PDF")
 
         self.percentage += 2
 
         # Convert to RGB pixel data
-        percentage_per_page = 45.0 / len(page_filenames)
-        for page in range(1, len(page_filenames) + 1):
-            pdf_filename = f"/tmp/page-{page}.pdf"
-            png_filename = f"/tmp/page-{page}.png"
-            rgb_filename = f"/tmp/page-{page}.rgb"
-            width_filename = f"/tmp/page-{page}.width"
-            height_filename = f"/tmp/page-{page}.height"
-            filename_base = f"/tmp/page-{page}"
-
-            self.update_progress(
-                f"Converting page {page}/{len(page_filenames)} to pixels"
-            )
-
-            # Convert to png
-            run_command(
-                ["pdftocairo", pdf_filename, "-png", "-singlefile", filename_base],
-                error_message="Conversion from PDF to PNG failed",
-                timeout_message=f"Error converting from PDF to PNG, pdftocairo timed out after {DEFAULT_TIMEOUT} seconds",
-            )
+        percentage_per_page = 45.0 / num_pages
+        page_base = "/tmp/page"
+        for page in range(1, num_pages + 1):
+            ppm_filename = f"{page_base}-{page}.ppm"
+            rgb_filename = f"{page_base}-{page}.rgb"
+            width_filename = f"{page_base}-{page}.width"
+            height_filename = f"{page_base}-{page}.height"
+            filename_base = f"{page_base}-{page}"
 
-            # Save the width and height
-            with Image.open(png_filename, "r") as im:
-                width, height = im.size
-            with open(width_filename, "w") as f:
-                f.write(str(width))
-            with open(height_filename, "w") as f:
-                f.write(str(height))
+            self.update_progress(f"Converting page {page}/{num_pages} to pixels")
 
-            # Convert to RGB pixels
+            # Convert to PPM, which is essentially an RGB format
             run_command(
```

Comment:
I need to lower the timeout on this one since it's now doing so per page instead of for the full document.

Reply:
No longer a problem.